libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookClientWrapper.cs

```diff
@@ -96,68 +90,52 @@ public async Task<FacebookClientWrapper> GetAPIAsync(Activity activity)
                 pageId = activity.From.Id;
             }
 
-            var token = await Options.GetAccessTokenForPageAsync(pageId).ConfigureAwait(false);
+            var token = await _options.GetAccessTokenForPageAsync(pageId).ConfigureAwait(false);
 
             if (string.IsNullOrWhiteSpace(token))
             {
                 throw new ArgumentException(nameof(token));
             }
 
-            return new FacebookClientWrapper(new FacebookAdapterOptions(Options.VerifyToken, Options.AppSecret, token));
+            return new FacebookClientWrapper(new FacebookAdapterOptions(_options.VerifyToken, _options.AppSecret, token));
         }
 
         /// <summary>
         /// Verifies the SHA1 signature of the raw request payload before bodyParser parses it will abort parsing if signature is invalid, and pass a generic error to response.
         /// </summary>
         /// <param name="request">An Http request object.</param>
-        /// <param name="response">An Http response object.</param>
-        /// <param name="cancellationToken">A cancellation token for the task.</param>
-        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
-        public async Task<bool> VerifySignatureAsync(HttpRequest request, HttpResponse response, CancellationToken cancellationToken)
+        /// <returns>The result of the comparison between the signature in the request and hashed body.</returns>
+        public virtual bool VerifySignature(HttpRequest request)
         {
             if (request == null)
             {
                 throw new ArgumentNullException(nameof(request));
             }
 
-            if (response == null)
-            {
-                throw new ArgumentNullException(nameof(response));
-            }
-
             var expected = request.Headers["x-hub-signature"];
 
-            string calculated = null;
-            using (var hmac = new HMACSHA256(Encoding.UTF8.GetBytes(Options.AppSecret)))
+            string calculated;
+
+            using (var hmac = new HMACSHA256(Encoding.UTF8.GetBytes(_options.AppSecret)))
             {
                 using (var bodyStream = new StreamReader(request.Body))
                 {
                     calculated = $"sha1={hmac.ComputeHash(Encoding.UTF8.GetBytes(bodyStream.ReadToEnd()))}";
                 }
             }
 
-            if (expected != calculated)
```

Comment:
This unauthorized response went nowhere? I don't see it in ProcessAsync either

Reply:
It's in ProcessAsync, in line 138.

Reply:
Saw it, my bad